code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -527,7 +527,15 @@
 		name = "drinking glass"
 		desc = "Your standard drinking glass."
 		return
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
```

Comment:
To check if the glass already has a name, you can cheat a little by just... checking the name. However, since this rename works for both glasses and shot glasses, you'll need to check for both.
```
if((type == /obj/item/weapon/reagent_containers/food/drinks/drinkingglass && name == "glass of ..what?") || (type == /obj/item/weapon/reagent_containers/food/drinks/drinkingglass/shotglass && name == "shot of... what?"))
```
Something along those lines will work. Also setting a variable to see if the glass was renamed would also be nice, since when you take a sip of the glass it will trigger `on_reagent_change()` and change the name to "glass of ..what?". A check for the variable before the renaming list comes in would prevent the automatic renaming. If the volume is 0, remove the variable though so it renames itself back to just `drinking glass`.